### ISSUE TITLE
Tutorial teaches inaccessible form input coding

### ISSUE THREAD

**Issue by jrpool** [angular/angular#31075]
# 📚 Docs or angular.io bug report

### Description

The tutorial at https://angular.io/start/forms#create-the-checkout-form teaches to place a "label" element above an "input" element but not to associate them with each other programmatically. That violates WCAG and causes assistive technologies to be unable to tell their users what the text inputs are for.

### What's the affected URL?**
https://angular.io/start/forms#create-the-checkout-form

## 🔥 Exception or Error
<pre><code>

</code></pre>

**Comment by brandonroberts** [comment]
@jrpool is the suggestion to update the labels with a `for` that points to the corresponding `input`? If so, do you want to put in a PR for this?

**Comment by jrpool** [comment]
@Yes, @brandonroberts, or else the simpler technique of moving the closing "label" tag to after the input. I have submitted a PR for that. Since I am only now learning Angular, I may have found the wrong component for this and I have not tried to construct any test for the PR, assuming that I would probably do it incorrectly.

**Comment by kapunahelewong** [comment]
Thank you, @jrpool for this. I try to correct accessibility issues when I see them in the examples, but am definitely still learning my way through. Would you like to collaborate on a PR? I'd be happy to help you find the right file and accompany you through the process.

I do have a PR in the works right now, https://github.com/angular/angular/pull/28571, involving the tutorial. After that merges, we could start a PR for your fix, if you'd like.

@jrpool, you might also be interested in @martiansnoop's PR, https://github.com/angular/angular/pull/28535.

**Comment by jrpool** [comment]
Thanks, @kapunahelewong. Yes, getting your collaboration would be very helpful.
The PR is https://github.com/angular/angular/pull/31213.
I see I have been asked to squash the commits and remessage. I'll plan to do that soon.

**Comment by kapunahelewong** [comment]
Closing this one out since #31213 landed. Thank you so much, @jrpool!

For reference, accessibility is still a project for the docs and https://github.com/angular/angular/pull/28535 is currently open. 🎉
